### PR TITLE
[BEAM-498] Port some example utils from OldDoFn to DoFn

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

R: @dhalperi

### CODE REVIEW

**Comment by kennknowles** [comment]
Jenkins got killed in the middle of testing this; rebasing to kick.

**Comment by dhalperi** [comment]
R: -@dhalperi @tgroh 

Thomas has touched this code most recently, is probably a better reviewer.

CC: @hadarg @francesperry  -- do these changes to mobile gaming example affect the example writeup at all?

File: examples/java8/src/main/java/org/apache/beam/examples/complete/game/GameStats.java [hunk 1]
```diff
@@ -193,21 +193,26 @@ public void processElement(ProcessContext c, BoundedWindow window) {
...
+            "STRING", (c, w) -> c.element().getKey()));
+    tableConfigure.put(
+        "total_score",
+        new WriteWindowedToBigQuery.FieldInfo<KV<String, Integer>>(
+            "INTEGER", (c, w) -> c.element().getValue()));
+    tableConfigure.put(
+        "window_start",
+        new WriteWindowedToBigQuery.FieldInfo<KV<String, Integer>>(
+            "STRING",
+            (c, window) -> {
```
Comment (tgroh) [inline examples/java8/src/main/java/org/apache/beam/examples/complete/game/GameStats.java]: your window parameters are inconsistent. I'd prefer the parameters to always be the same, and you can explicate if you need to downcast to a more precise type.

Comment (kennknowles) [inline examples/java8/src/main/java/org/apache/beam/examples/complete/game/GameStats.java]: Done, here and throughout.

**Pull request closed without merging**